lib/vagrant/util/platform.rb

```diff
@@ -139,8 +139,12 @@ def windows_hyperv_enabled?
           return @_windows_hyperv_enabled if defined?(@_windows_hyperv_enabled)
 
           @_windows_hyperv_enabled = -> {
-            ["Get-WindowsOptionalFeature", "Get-WindowsFeature"].each do |cmd_name|
-              ps_cmd = "$(#{cmd_name} -FeatureName Microsoft-Hyper-V-Hypervisor).State"
+            {:"Get-WindowsOptionalFeature" => ["-Online"], :"Get-WindowsFeature" => []}.each do |cmd_name, arguments|
+              parameters = arguments.join " "
+              ps_cmd = "$(#{cmd_name} -FeatureName Microsoft-Hyper-V-Hypervisor #{parameters}).State"
+              if cmd_name == "Get-WindowsFeature"
```

Comment:
I don't think this part is required since a failure is handled by `Vagrant::Util::PowerShell.execute_cmd`

Reply:
Would it be required if HVM Microsoft HVM installed or not, if installed it should fail the vagrant because of that

Reply:
Ya, if it fails it shouldn't raise an error. This is the [code for the powershell execute_cmd](https://github.com/hashicorp/vagrant/blob/master/lib/vagrant/util/powershell.rb#L116). It returns `nil` if the return code is not 0. Testing this theory out (with a little bit of an extreme example). For a diff like

```
diff --git a/lib/vagrant/util/platform.rb b/lib/vagrant/util/platform.rb
index 1d8e32119..57146031f 100644
--- a/lib/vagrant/util/platform.rb
+++ b/lib/vagrant/util/platform.rb
@@ -142,6 +142,7 @@ module Vagrant
             check_commands = Array.new.tap do |c|
               c << "(Get-WindowsOptionalFeature -FeatureName Microsoft-Hyper-V-Hypervisor -Online).State"
               c << "(Get-WindowsFeature -FeatureName Microsoft-Hyper-V-Hypervisor).State"
+              c << "(SOMETHING BROKEN).state"
             end
``` 

Vagrant does the correct thing.